components/test-file-results.html

```diff
@@ -0,0 +1,168 @@
+              <td class$="result [[ _subTestResultForTestRun(testRun, subtestName) ]]">
+                {{ _subTestResultForTestRun(testRun, subtestName) }}
+              </td>
+            </template>
+          </tr>
+        </template>
+
+      </tbody>
+    </table>
+  </template>
+
+  <script>
+    class TestFileResults extends window.Polymer.Element {
+      static get is () { return 'test-file-results' }
+
+      static get properties () {
+        return {
+          testRuns: {
+            type: Array
+          },
+          testFile: {
+            type: String,
+            observer: '_testFileChanged'
+          },
+          subTestNames: {
+            type: Array,
+            value: []
+          }
+        }
+      }
+
+      connectedCallback () {
+        super.connectedCallback()
+        console.assert(this.testFile)
+        console.assert(this.testFile[0] === '/')
+        console.assert(this.testRuns)
+        console.assert(this.testRuns.length > 0)
+      }
+
+      async fetchTestFile () {
+        const resultPerTestRun = await Promise.all(this.testRuns.map(tr => this.loadResultFile(tr)))
+
+        resultPerTestRun.forEach((resultData, i) => {
+          if (!resultData) {
+            this.testRuns[i].subtests = {}
+            this.testRuns[i].subtests['Test file'] = { status: '(results not found)' }
+            return
+          }
+          this.testRuns[i].subtests = {}
+          this.testRuns[i].subtests['Test file'] = { status: resultData.status }
+
+          if (!this.subTestNames.includes('Test file')) {
+            this.subTestNames = this.subTestNames.concat(['Test file'])
+          }
+
+          for (let subTestResult of resultData.subtests) {
+            this.testRuns[i].subtests[subTestResult.name] = subTestResult
+            if (!this.subTestNames.includes(subTestResult.name)) {
+              this.subTestNames = this.subTestNames.concat([subTestResult.name])
+            }
+          }
+        })
+      }
+
+      async loadResultFile (testRun) {
+        const url = this.resultsURL(testRun, this.testFile)
+        const response = await window.fetch(url)
+        if (response.status !== 200) {
+          console.error('Got non-200 status for url:', url)
+          return Promise.resolve(null)
+        }
+        return response.json()
+      }
+
+      _testFileChanged () {
+        this.subTestNames = []
+        this.fetchTestFile()
+      }
+
+      resultsURL (testRun, testFile) {
+        // This is relying on the assumption that result files are under a directory named SHA[0:10]
+        const resultsBase = testRun.results_url.split('/' + testRun.revision)[0]
+        const platformID = `${testRun.browser_name}-${testRun.browser_version}-${testRun.os_name}-${testRun.os_version}`
+        return `${resultsBase}/${testRun.revision}/${platformID}${testFile}`
+      }
+
+      _subTestResultForTestRun (testRun, subtestName) {
```

Comment:
Nit: For consistency, you could use subtestResult... (here and elsewhere) instead of subTestResult... since elsewhere subtests is treated as one word.